mne/channels/layout.py

```diff
@@ -931,3 +931,98 @@ def apply_montage(info, montage):
         raise ValueError('None of the sensors defined in the montage were '
                          'found in the info structure. Check the channel '
                          'names.')
+
+
+def generate_2d_layout(xy, w=.07, h=.05, pad=.02, ch_names=None,
+                       ch_indices=None, name='ecog', bg_image=None):
+    '''Generates a 2-D layout for plotting with plot_topo methods and
+    functions. XY points will be normalized between 0 and 1, where
+    normalization extremes will be either the min/max of xy, or
+    the width/height of bg_image.
+
+    Parameters
+    -------
+    xy : ndarray (N x 2) | 'interactive'
+        The xy coordinates of sensor locations. If 'interactive',
+        bg_image must not be None, as this will be displayed
+        and the user must click on positions of the image to
+        create xy values.
+
+    w, h : float
+        The width and height of each sensor's axis (between 0 and 1)
+
+    pad : float
+        Portion of the box to reserve for padding. The value can range between
+        0.0 (boxes will touch, default) to 1.0 (boxes consist of only padding).
+
+    ch_names : list
+        The names of each channel. Must be a list of strings, with one
+        string per channel.
+
+    ch_indices : list
+        Index of each channel - must be a collection of unique integers,
+        one index per channel.
+
+    name : string
+        The name of this layout type.
+
+    bg_image : str | ndarray
+        The image over which sensor axes will be plotted. Either a path to an
+        image file, or an array that can be plotted with plt.imshow. If
+        provided, xy points will be normalized by the width/height of this
+        image. If not, xy points will be normalized by their own min/max.
+
+
+    Returns
+    --------
+    layout : Layout
+        A Layout object that can be plotted with plot_topo
+        functions and methods.
+    '''
+
+    from matplotlib.pyplot import imread
+
+    # Check that len(ch_names) == len(ch_indices) if not None
+    # Check that len(ch_names) == len(xy) if not None
+
+    if ch_indices is None:
+        ch_indices = np.arange(xy.shape[0])
+    if ch_names is None:
+        ch_names = ['{0}'.format(i) for i in ch_indices]
+
+    # To interactively pull xy values
+    if xy == 'interactive' and bg_image is not None:
+        # Pull xy coordinates interactively
+        # Can add this later but the function should work with pre-suppleid
+        # xy points first
+        pass
```

Comment:
don't write code that "may" be useful in the future. Or let's keep this PR as WIP and merge when you're happy about the new feature